pkg/image/validation/registry.go

```diff
@@ -0,0 +1,97 @@
+package validation
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/combustion"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+)
+
+const (
+	registryComponent = "Artifact Registry"
+)
+
+func validateEmbeddedArtefactRegistry(ctx *image.Context) []FailedValidation {
+	var failures []FailedValidation
+	def := ctx.ImageDefinition
+
+	if combustion.IsEmbeddedArtifactRegistryEmpty(def.EmbeddedArtifactRegistry) {
+		return failures
+	}
```

Comment:
Can we drop this check? Relying on the combustion package for the validation itself seems weird and trying to iterate over nil slices should not lead to errors below.

Reply:
I didn't love this dependency either; it was mostly because the check is used at both runtime and validation time. But since this is the only instance where that happens (so far) and like you said it doesn't affect the behavior, I agree with dropping it.